### PR TITLE
Fast wrappers for ImagePlus-Img-conversions

### PR DESCRIPTION
**Pull request by maarzt**
This PR is required to improve the IJ1-IJ2 compatibility layer. It adds wrappers to translate between ImagePlus and ImgPlus. These new wrappers are faster than the existing ones. This is achieved by sharing buffers instead of copying and being lazily.

Add VirtualStackAdapter: 
* It's similar to ImagePlusAdapter as it presents an IJ1 ImagePlus as an IJ2 PlanarImg, but it's way faster during initializing when used for an ImagePlus that contains a VirtualStacks.

Refactor ImageJVirtualStack: 
* Code quality improvements.
* Fixing a bug when multiple ImageProcessors are used simultaneously.
* Adding ImgToVirtualStack#wrap method which simply takes an Img(Plus) and returns an ImagePlus.

Add PlanarImgToVirtualStack:
* Similar to ImageJVirtualStack but not as flexible. It shares buffers between the PlanarImg and ImagePlus instead of copying the planes.

Add ArrayImgToVirtualStack
* Similar to PlanarImgToVirtualStack, but works for 2d ArrayImg.

### CODE REVIEW

File: src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ protected ImageJVirtualStack(final RandomAccessibleInterval< S > source, final C
 	protected ImageJVirtualStack(final RandomAccessibleInterval< S > source, final Converter< S, T > converter,
 			final T type, final int ijtype, ExecutorService service)
 	{
-		super( (int) source.dimension( 0 ), getDimension1Size( source ), null, null );
+		super( (int) source.dimension( 0 ), (int) source.dimension( 1 ), null, null );
```
Comment (tpietzsch) [inline src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java]: This is wrong. Now it will not work for 1D images anymore.

File: src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java [hunk 2]
```diff
@@ -92,58 +99,57 @@ protected ImageJVirtualStack(final RandomAccessibleInterval< S > source, final C
...
+		this.max = max;
+	}
+
+	private static < S > RandomAccessibleInterval< S > zeroMin( RandomAccessibleInterval< S > source )
 	{
-		return Views.isZeroMin( source ) ? source : Views.zeroMin( source );
+		if ( Views.isZeroMin( source ) )
+			return source;
+		Interval zeroMinInterval = new FinalInterval( Intervals.dimensionsAsLongArray( source ) );
+		return Views.interval( Views.zeroMin( source ), zeroMinInterval );
```
Comment (tpietzsch) [inline src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java]: This (`Views.interval`) is unnecessary. `Views.zeroMin` already does it. It was fine before.
Actually, the whole `return Views.isZeroMin( source ) ? source : ...;` should be done in `Views.zeroMin()`, so that we can get rid of this zeroMin method here altogether

File: src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStackFloat.java [hunk 3]
```diff
@@ -101,7 +106,7 @@ public void setMinMax(final RandomAccessibleInterval< S > source, final Converte
 
 			System.out.println( "fmax = " + max );
 			System.out.println( "fmin = " + min );
```
Comment (tpietzsch) [inline src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStackFloat.java]: Also not part of this PR, but these `println`s should be removed

File: src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStackFloat.java [hunk 4]
```diff
@@ -190,7 +195,21 @@ public Void call() throws Exception
 
 		System.out.println( "fmax = " + max );
 		System.out.println( "fmin = " + min );
```
Comment (tpietzsch) [inline src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStackFloat.java]: Also not part of this PR, but these `println`s should be removed

File: src/main/java/net/imglib2/img/display/imagej/PlanarImgToVirtualStack.java [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+		if ( !( img instanceof PlanarImg ) )
+			throw new IllegalArgumentException( "Image must be a PlanarImg." );
+		IntUnaryOperator indexer = getIndexer( imgPlus );
+		VirtualStack stack = new PlanarImgToVirtualStack( ( PlanarImg< ?, ? > ) img, indexer );
+		ImagePlus imagePlus = new ImagePlus( imgPlus.getName(), stack );
+		imagePlus.setDimensions( dimension( imgPlus, Axes.CHANNEL ), dimension( imgPlus, Axes.Z ), dimension( imgPlus, Axes.TIME ) );
+		return imagePlus;
+	}
+
+	private static int dimension( ImgPlus imgPlus, AxisType axisType )
```
Comment (tpietzsch) [inline src/main/java/net/imglib2/img/display/imagej/PlanarImgToVirtualStack.java]: `ImgPlus<?>`

File: src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java (hunk 1 above)
Comment (maarzt) [reply src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java]: It didn't work with 1d images before. The strategy for 1d images is to add an axis and the use ImageJVirtualStack.

**Comment by maarzt** [comment]
@tpietzsch I finished integrating the changes you suggested in the review.

**Comment by tpietzsch** [comment]
@maarzt `ImagePlusImg` constructors still use now-deprecated (by you :-) `PlanarImg` constructors. If you find some time, it would be great if you could fix that...

**Comment by tpietzsch** [comment]
Releasing `2.0.0-beta-41` now ...
